### ISSUE TITLE
Content-Digest verification PR

### ISSUE THREAD

**Issue by jschlyter** [pyauth/http-message-signatures#9]
I'm looking at Content-Digest verification. Would it make sense to implement this as a helper in this library?

**Comment by kislyuk** [comment]
Yes. There is a bit of tech debt: the other library that I maintain that depends on this one already implements content-digest verification (https://github.com/pyauth/requests-http-signature/blob/develop/requests_http_signature/__init__.py#L273-L288) but it is specific to the Requests API. I would like to move that code to a helper in this library and wire up the other one to use it.

PRs are welcome.
